core/src/main/java/org/apache/calcite/runtime/CompressionFunctions.java

```diff
@@ -62,4 +65,30 @@ public static ByteString compress(String data) {
     }
   }
 
+  /**
+   * MySql Decompression is based on zlib.
+   * <a href="https://docs.oracle.com/javase/8/docs/api/java/util/zip/Inflater.html">Inflater</a>
+   * is used to implement decompression.
+   */
+  public static String uncompress(ByteString data) {
+    try {
+      if (data == null) {
+        return null;
+      } else if (data.length() == 0) {
+        return "";
+      }
+      InflaterInputStream inflaterStream = new InflaterInputStream(
+          new ByteArrayInputStream(data.getBytes(), 4, data.length()));
+      return IOUtils.toString(inflaterStream);
+    } catch (IOException e) {
+      return null;
+    }
+  }
+
+  /**
+   * MySQl Supports uncompress function for string which returns NULL value.
+   */
+  public static String uncompress(String data) {
+    return null;
+  }
```

Comment:
This is not necessary, Calcite supports implicit type coercion, if you declare to support `OperandTypes.BINARY` only and pass a `STRING` actually, Calcite would try to coerce the `STRING` to binary before passing through to the runtime code.

Reply:
@danny0405 Can you please look for attached error, and help me understand the issue.

Reply:
`Error while executing SQL "select UNCOMPRESS(p0) from (values ('test')) as t(p0)": Error while compiling generated Java code:
public org.apache.calcite.linq4j.Enumerable bind(final org.apache.calcite.DataContext root) {
  final org.apache.calcite.linq4j.Enumerable _inputEnumerable = org.apache.calcite.linq4j.Linq4j.asEnumerable(new String[] {
    "test"});
  return new org.apache.calcite.linq4j.AbstractEnumerable(){
      public org.apache.calcite.linq4j.Enumerator enumerator() {
        return new org.apache.calcite.linq4j.Enumerator(){
            public final org.apache.calcite.linq4j.Enumerator inputEnumerator = _inputEnumerable.enumerator();
            public void reset() {
              inputEnumerator.reset();
            }

            public boolean moveNext() {
              return inputEnumerator.moveNext();
            }

            public void close() {
              inputEnumerator.close();
            }

            public Object current() {
              return org.apache.calcite.runtime.CompressionFunctions.uncompress((org.apache.calcite.avatica.util.ByteString) (inputEnumerator.current() == null ? (String) null : inputEnumerator.current().toString()));
            }

          };
      }

    };
}


public Class getElementType() {
  return java.lang.String.class;
}


];`

Reply:
> This is not necessary, Calcite supports implicit type coercion, if you declare to support `OperandTypes.BINARY` only and pass a `STRING` actually, Calcite would try to coerce the `STRING` to binary before passing through to the runtime code.

Also Implicit type coercion from binary to string is not supported as of now